### PR TITLE
Load game data from files

### PR DESCRIPTION
**Pull request by fraillt**
Hello,
This includes multiple changes:
1) updated cargo.toml file, to only use amethyst features that are required (much faster build time).
2) load items, enemies and consumables from ./assets/data/* files at startup.
3) updated item spawn logic, now accepts list of items with probabilities.

that's pretty much it ;)

### CODE REVIEW

**Comment by cdsupina** [comment]
I did a quick skim of the changes and overall I like this, however, I noticed when I ran the game that no enemies were spawning even though the `phase_map` in the gamemaster was unchanged. Do you know why this is?

**Comment by fraillt** [comment]
Really strange, it would sound crazy, but it worked for me... maybe I accidentally missed some file or something during git commit... I'll check it tomorrow in the morning

**Comment by cdsupina** [comment]
I'll have time tomorrow too. I'll take a closer look then. Just to be clear when you run it now the enemies are spawning after 8 ticks?

**Comment by fraillt** [comment]
It turns out I incorrectly understood serde(default) field attribute :) and I loaded wrong defaults for items, enemies and consumables. now everything works.

**Comment by cdsupina** [comment]
Great. I'll look it over and then merge. I also plan on implementing a store system next which will lead to item spawns, but that is fine for now.

**Comment by cdsupina** [comment]
There is one more issue. I encountered it before when I was cleaning up the compiler warnings. In `collision_handler.rs` you removed the `ReadStorage<'s, Transform>` from the system data. For some reason when you remove this line the collisions don't work right. Notice how if you run the game the spaceship now collides through the spaceship instead of bouncing off of it. I don't completely understand why this is, but I think we should add that back in or figure out why this is happening before merging. I'll comment the lines in review.

File: src/systems/collision_handler.rs [hunk 1]
```diff
@@ -26,7 +25,6 @@ impl<'s> System<'s> for CollisionHandlerSystem {
         Read<'s, EventChannel<CollisionEvent>>,
         WriteStorage<'s, Spaceship>,
         WriteStorage<'s, Enemy>,
-        ReadStorage<'s, Transform>,
```
Comment (cdsupina) [inline src/systems/collision_handler.rs]: Removing this breaks collisions for a reason unknown to me.

File: assets/data/consumables.ron [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "defense": (
+        defense_value: 70,
+        speed: 10,
```
Comment (cdsupina) [inline assets/data/consumables.ron]: Defense wrench moves super slow. Remove speed attribute here so it moves the default speed?

File: src/systems/collision_handler.rs (hunk 1 above)
Comment (fraillt) [reply src/systems/collision_handler.rs]: the problem was that there was no dependency relationship between collision detection and collision handling, adding this dependency solved the issue.

File: assets/data/consumables.ron (hunk 2 above)
Comment (fraillt) [reply assets/data/consumables.ron]: restored to default.

**Comment by fraillt** [comment]
Thanks ;)

**Pull request merged**